src/briefcase/console.py

```diff
@@ -302,24 +304,72 @@ def __init__(self, printer=Printer(), enabled=True):
         self.print = printer
         self.input = printer.console.input
         self.enabled = enabled
-        self._wait_bar: Progress = None
+
         # Signal that Rich is dynamically controlling the console output.
-        # Therefore, all output must be printed to the screen by Rich to
-        # prevent corruption of dynamic elements like the Wait Bar.
+        # Therefore, all output must be printed to the console by Rich to
+        # prevent corruption by dynamic elements like the Wait Bar.
         self.is_output_controlled = False
 
-    def prompt(self, *values, markup=False, **kwargs):
-        """Print to the screen for soliciting user interaction.
+        self._live_display: Live = None
+        self._live_display_stack = []
+        self._wait_bar: Progress = None
 
-        :param values: strings to print as the user prompt
-        :param markup: True if prompt contains Rich markup
+    def _live_display_add(self, renderable):
+        """Add a renderable to the top of the Live Display."""
+        if renderable not in self._live_display_stack:
+            self._live_display_stack.append(renderable)
+            self._live_display_update()
+
+    def _live_display_remove(self, renderable):
+        """Remove a renderable from the Live Display."""
+        with contextlib.suppress(ValueError):
+            self._live_display_stack.remove(renderable)
+        self._live_display_update()
+        # Print non-transient renderables so their final refresh remains in
+        # the console. When a renderable is removed from the Live Display,
+        # it is also completely removed from the console.
+        if not renderable.live.transient:
+            self.print(renderable)
+
+    def _live_display_update(self):
+        """Update the Live Display with the current stack of renderables."""
+        if not self._live_display:
+            self._live_display = Live(
+                refresh_per_second=15,
+                transient=True,
+                console=self.print.console,
+            )
+        if self._live_display_stack:
+            self.is_output_controlled = True
+            self._live_display.update(Group(*self._live_display_stack[::-1]))
+            self._live_display.start()
+        else:
+            self._live_display.stop()
+            self.is_output_controlled = False
+
+    def remove_dynamic_elements(self):
```

Comment:
I didn't implement this idea....but this could also set a "do not allow starting the live display" flag here that would be unset in `restore_dynamic_elements()`. This would allow adding renderables but they wouldn't be displayed until the restore call.

Given the use cases, it seems reasonable that a new renderable will be added after calling remove.